undertow/src/main/java/org/wildfly/extension/undertow/UndertowTransformers.java

```diff
@@ -0,0 +1,85 @@
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.wildfly.extension.undertow;
+
+import static org.wildfly.extension.undertow.ListenerResourceDefinition.READ_TIMEOUT;
+import static org.wildfly.extension.undertow.ListenerResourceDefinition.WRITE_TIMEOUT;
+
+import org.jboss.as.controller.ModelVersion;
+import org.jboss.as.controller.transform.ExtensionTransformerRegistration;
+import org.jboss.as.controller.transform.SubsystemTransformerRegistration;
+import org.jboss.as.controller.transform.description.AttributeTransformationDescriptionBuilder;
+import org.jboss.as.controller.transform.description.ChainedTransformationDescriptionBuilder;
+import org.jboss.as.controller.transform.description.DiscardAttributeChecker;
+import org.jboss.as.controller.transform.description.RejectAttributeChecker.SimpleRejectAttributeChecker;
+import org.jboss.as.controller.transform.description.ResourceTransformationDescriptionBuilder;
+import org.jboss.as.controller.transform.description.TransformationDescriptionBuilder;
+import org.jboss.dmr.ModelNode;
+
+/**
+ * @author <a href="mailto:ropalka@redhat.com">Richard Opalka</a>
+ */
+public class UndertowTransformers implements ExtensionTransformerRegistration {
+    private static final ModelVersion MODEL_VERSION_WILDFLY_23 = ModelVersion.create(12, 0, 0);
+
+    @Override
+    public String getSubsystemName() {
+        return UndertowExtension.SUBSYSTEM_NAME;
+    }
+
+    @Override
+    public void registerTransformers(SubsystemTransformerRegistration subsystemRegistration) {
+
+        ChainedTransformationDescriptionBuilder chainedBuilder = TransformationDescriptionBuilder.Factory.createChainedSubystemInstance(subsystemRegistration.getCurrentSubsystemVersion());
+
+        registerTransformersWildFly23(chainedBuilder.createBuilder(subsystemRegistration.getCurrentSubsystemVersion(), MODEL_VERSION_WILDFLY_23));
+
+        chainedBuilder.buildAndRegister(subsystemRegistration, new ModelVersion[]{MODEL_VERSION_WILDFLY_23});
+    }
+
+    private static void registerTransformersWildFly23(ResourceTransformationDescriptionBuilder subsystemBuilder) {
+        final ResourceTransformationDescriptionBuilder serverBuilder = subsystemBuilder.addChildResource(UndertowExtension.SERVER_PATH);
+
+        final AttributeTransformationDescriptionBuilder http = serverBuilder.addChildResource(UndertowExtension.HTTP_LISTENER_PATH).getAttributeBuilder()
+                .setDiscard(DiscardAttributeChecker.DEFAULT_VALUE, WRITE_TIMEOUT)
+                .setDiscard(DiscardAttributeChecker.DEFAULT_VALUE, READ_TIMEOUT)
+                .addRejectCheck(new SimpleRejectAttributeChecker(ModelNode.TRUE), WRITE_TIMEOUT.getName())
+                .addRejectCheck(new SimpleRejectAttributeChecker(ModelNode.TRUE), READ_TIMEOUT.getName());
```

Comment:
This transformation does not look correct to me.  A new default value should not result in any rejections, but rather use an AttributeConverter.
I've fixed this (and included missing tests) in this PR: https://github.com/wildfly/wildfly/pull/16455